package/MDAnalysis/analysis/base.py

```diff
@@ -430,25 +471,231 @@ def run(self, start=None, stop=None, step=None, frames=None,
+                    np.array_split(frames, n_bslices),
+                    )
+                   ]
+
+        self._bslices = slices
+
+        return self._bslices
+
+
+    def _setup_scheduler(self, scheduler, n_workers):
+        """
+        Configure parameters necessary for running a distributed workload.
+
+        Parameters
+        ----------
+        scheduler : str, optional 
+            scheduling type:
+              - 'dask' implies `dask` scheduler (need to install MDAnalysis[dask] for that to work)
+              - 'multiprocessing' works with standard python multiprocessing module
+              - None if you want to run things locally
+        n_workers : int, optional
+            number of workers (local or remote processes).
+        """
+        if scheduler is None:
+            n_workers = 1
+            kwargs = {}
+        elif scheduler == 'multiprocessing' or scheduler == 'dask':
+            kwargs = {'n_workers':n_workers, 'scheduler':'processes'}
+        else:
+            raise ValueError("`scheduler` should be either 'dask', 'multiprocessing' or None")
+
+        self._scheduler_kwargs = kwargs
+        self._n_bslices = n_workers
+
+    def run(self, start=None, stop=None, step=None, frames=None,
+            verbose=None, n_workers=1, 
+            *, scheduler=None,
+            progressbar_kwargs={},
+            ):
+        """Perform the calculation
+
+        Parameters
+        ----------
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+
+            .. versionadded:: 2.2.0
+
+        verbose : bool, optional
+            Turn on verbosity
+        
+        progressbar_kwargs : dict, optional
+            ProgressBar keywords with custom parameters regarding progress bar position, etc; 
+            see :class:`MDAnalysis.lib.log.ProgressBar` for full list.
+
+        scheduler : str, optional 
+            Enables parallel execution of :meth:`AnalysisBase.run`.
+            Available schedulers
+              - 'dask' implies execution using `dask` processes (need to install MDAnalysis[dask] for that to work)
+              - 'multiprocessing' works with standard python multiprocessing module
+              - `None` if you want to run things locally
+        n_workers : int, optional
+            number of workers. 
+            Will become number of processes in multiprocessing.Pool(...) with `multiprocessing` scheduler,
+            and `n_workers` with `dask` scheduler.
```

Comment:
Following the above, do these need a `.. versionadded:: 2.6.0` (or whatever version will be relevant)?

Reply:
I think after the code part is done and agreed upon, I'll go through the documentation and add `versionadded` in proper places.